pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -274,19 +274,27 @@ def __init__(
         self,
         coeffs,
         observables,
+        reorder_fn=None,
         pulses: List["HardwarePulse"] = None,
         settings: Union["RydbergSettings", "TransmonSettings"] = None,
     ):
         self.settings = settings
         self.pulses = [] if pulses is None else pulses
+
+        def trivial_func(_, y):
+            return y
+
+        self.reorder_fn = trivial_func if reorder_fn is None else reorder_fn
         super().__init__(coeffs, observables)
 
     def __call__(self, params, t):
-        params = _reorder_parameters(params, self.coeffs_parametrized)
+        params = self.reorder_fn(params, self.coeffs_parametrized)
         return super().__call__(params, t)
 
     def __add__(self, other):
         if isinstance(other, HardwareHamiltonian):
+            # if not self.reorder_fn == other.reorder_fn:
+            #     raise ValueError(f"Cannot add two HardwareHamiltonians with different reorder functions. Received reorder_fns {self.reorder_fn} and {other.reorder_fn}")
```

Comment:
Is there a good way to compare two callables?

Reply:
It's really annoying to compare callables. This is dev facing functionality, so I think we can make it a point to use references to the same callable if two callables are meant to be equivalent, in which case this would work.

Reply:
This doesn't work? If they both refer to the exact same function (which they should if they match), I thought this would return True. And otherwise False.

Reply:
Right! Dont know what I was missing earlier! thanks :)